patches/net/minecraft/client/multiplayer/ClientCommonPacketListenerImpl.java.patch

```diff
@@ -15,3 +21,45 @@
  
      protected abstract RegistryAccess.Frozen registryAccess();
  
+@@ -189,6 +_,10 @@
+     }
+ 
+     public void send(Packet<?> p_295097_) {
++        if (!net.neoforged.neoforge.network.registration.NetworkRegistry.getInstance().canSendPacket(p_295097_, this)) {
++            return;
++        }
++        
+         this.connection.send(p_295097_);
+     }
+ 
+@@ -196,6 +_,9 @@
+     public void onDisconnect(Component p_295485_) {
+         this.telemetryManager.onDisconnect();
+         this.minecraft.disconnect(this.createDisconnectScreen(p_295485_));
++        if (!this.connection.isMemoryConnection()) {
++            net.neoforged.neoforge.registries.RegistryManager.revertToFrozen();
++        }
+         LOGGER.warn("Client disconnected with reason: {}", p_295485_.getString());
+     }
+ 
+@@ -315,5 +_,20 @@
+         @OnlyIn(Dist.CLIENT)
+         static record PendingRequest(UUID id, URL url, String hash) {
+         }
++    }
++    
++    @Override
++    public Connection getConnection() {
++        return connection;
++    }
++    
++    @Override
++    public net.minecraft.util.thread.ReentrantBlockableEventLoop<?> getMainThreadEventLoop() {
```

Comment:
Given this just returns `minecraft` do we want to consider moving this default impl into the extension interface and have it just call the `getMinecraft` method the extension is adding?